loglist/findlog/main.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The findlog binary attempts to provide information about a log based on
+// ID or name.
+package main
+
+import (
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/google/certificate-transparency-go/loglist"
+)
+
+var (
+	logList = flag.String("log_list", loglist.LogListURL, "Location of master log list (URL or filename)")
+	verbose = flag.Bool("verbose", false, "Print more information")
+)
+
+func readPossibleURL(target string) ([]byte, error) {
+	u, err := url.Parse(target)
+	if err != nil || (u.Scheme != "http" && u.Scheme != "https") {
```

Comment:
Just a thought. If you removed the u.Scheme checks would the resulting error still be useful enough?

Reply:
The `u.Scheme` checks are there because `url.Parse("/path/to/file.json")` doesn't actually fail -- it still produces a `url.URL`.  

Checking for `u.Scheme == ""` instead might work, but I wonder what would happen if a misguided Windows user fed in `C:\myloglist.json` ...

Reply:
OK I don't think the Go HTTP library is great here. It would be better if you could parse a URL, pass the URL it to the Get and have it just work for whatever scheme handlers existed.